From c9bcccb3d05e5433128593638a573980eb3ba308 Mon Sep 17 00:00:00 2001
From: Palo Kisa <palo.kisa@gmail.com>
Date: Wed, 16 Nov 2016 09:04:06 +0000
Subject: Bug 372223: Fix crash on exit

The QCoreApplication instance wasn't deleted before libraries
were unloaded, which resulted in a crash during clean-up. See
https://github.com/lxde/lxqt/issues/1193#issuecomment-260337417.
---
 src/main.cpp | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

--- a/kalarm/main.cpp
+++ b/kalarm/main.cpp
@@ -30,6 +30,7 @@
 #include <KLocalizedString>
 
 #include <QDir>
+#include <QScopedPointer>
 
 #include <stdlib.h>
 
@@ -40,7 +41,10 @@ int main(int argc, char* argv[])
     KAlarmMigrateApplication migrate;
     migrate.migrate();
 
-    KAlarmApp* app = KAlarmApp::create(argc, argv);
+    // Use QScopedPointer to ensure the QCoreApplication instance is deleted
+    // before libraries unload, to avoid crashes during clean-up.
+    QScopedPointer<KAlarmApp> app(KAlarmApp::create(argc, argv));
+
     QStringList args = app->arguments();
     app->setAttribute(Qt::AA_UseHighDpiPixmaps, true);
 #if QT_VERSION >= 0x050600
